core/io/resource.cpp

```diff
@@ -124,6 +124,9 @@ String Resource::generate_scene_unique_id() {
 	return id;
 }
 
+/// warning: there is no validation here
+/// use is_ascii_identifier_char() to validate before passing it in
+/// example: if you pass in a quote, it will corrupt the scene when saving
```

Comment:
I must mention first that all new comments should follow the [comment style guide](https://docs.godotengine.org/en/stable/contributing/development/code_style_guidelines.html#comment-style-guide), ideally. So it should look more like this.
```suggestion
// Warning: there is no validation here.
// Use is_ascii_identifier_char() to validate before passing it in.
// Example: if you pass in a quote, it will corrupt the scene when saving.
```

But more importantly to me, the comment does summarise another reason as to why I was skeptical on exposing this. The outside developer should be warned about this:
- It should be documented in the class reference.
- It should not even be possible to do this. In this situation, a warning should appear, the invalid characters should be replaced or the setter should fail outright. Whatever is the most desired.

This would mean making a distinction between the `set_scene_unique_id` used internally (since the additional checks would be a waste), and the exposed `set_scene_unique_id`.

I believe that in this situation a lot of classes prefix the internal method with an underscore.